processor/samplingprocessor/tailsamplingprocessor/processor.go

```diff
@@ -244,37 +241,48 @@ func (tsp *tailSamplingSpanProcessor) makeDecision(id pdata.TraceID, trace *samp
 }
 
 // ConsumeTraceData is required by the SpanProcessor interface.
-func (tsp *tailSamplingSpanProcessor) ConsumeTraces(_ context.Context, td pdata.Traces) error {
+func (tsp *tailSamplingSpanProcessor) ConsumeTraces(ctx context.Context, td pdata.Traces) error {
 	tsp.start.Do(func() {
 		tsp.logger.Info("First trace data arrived, starting tail_sampling timers")
 		tsp.policyTicker.Start(1 * time.Second)
 	})
-
-	octds := internaldata.TraceDataToOC(td)
-	var errs []error
-	for _, octd := range octds {
-		if err := tsp.processTraces(octd); err != nil {
-			errs = append(errs, err)
+	resourceSpans := td.ResourceSpans()
+	for i := 0; i < resourceSpans.Len(); i++ {
+		resourceSpan := resourceSpans.At(i)
+		if resourceSpan.IsNil() {
+			continue
 		}
+		tsp.processTraces(resourceSpan)
 	}
-	return componenterror.CombineErrors(errs)
+	return nil
 }
 
-func (tsp *tailSamplingSpanProcessor) processTraces(td consumerdata.TraceData) error {
-
-	// Groupd spans per their traceId to minimize contention on idToTrace
-	idToSpans := make(map[traceKey][]*tracepb.Span)
-	for _, span := range td.Spans {
-		if len(span.TraceId) != 16 {
-			tsp.logger.Warn("Span without valid TraceId", zap.String("SourceFormat", td.SourceFormat))
+func (tsp *tailSamplingSpanProcessor) groupSpansByTraceKey(resourceSpans pdata.ResourceSpans) map[traceKey][]*pdata.Span {
+	idToSpans := make(map[traceKey][]*pdata.Span)
+	ilss := resourceSpans.InstrumentationLibrarySpans()
+	for j := 0; j < ilss.Len(); j++ {
+		ils := ilss.At(j)
+		if ils.IsNil() {
 			continue
 		}
-		tk := traceKey(span.TraceId)
-		idToSpans[tk] = append(idToSpans[tk], span)
+		spansLen := ils.Spans().Len()
+		for k := 0; k < spansLen; k++ {
+			span := ils.Spans().At(k)
+			tk := traceKey(span.TraceID().Bytes())
+			if len(tk) != 16 {
+				tsp.logger.Warn("Span without valid TraceId")
+			}
+			idToSpans[tk] = append(idToSpans[tk], &span)
+		}
 	}
+	return idToSpans
+}
+
+func (tsp *tailSamplingSpanProcessor) processTraces(resourceSpans pdata.ResourceSpans) error {
+	// Groupd spans per their traceId to minimize contention on idToTrace
```

Comment:
If you are updating the PR for some other reason, you might want to fix this typo: `s/Groupd/Group`

Reply:
Done